packages/components/src/utils/element-rect.ts

```diff
@@ -134,14 +134,17 @@ const POLL_RATE = 100;
  * milliseconds until it succeeds.
  */
 export function useTrackElementOffsetRect(
-	targetElement: HTMLElement | undefined | null
+	targetElement: HTMLElement | undefined | null,
+	deps: unknown[] = []
 ) {
 	const [ indicatorPosition, setIndicatorPosition ] =
 		useState< ElementOffsetRect >( NULL_ELEMENT_OFFSET_RECT );
 	const intervalRef = useRef< ReturnType< typeof setInterval > >();
 
 	const measure = useEvent( () => {
-		if ( targetElement ) {
+		// Check that the targetElement is still attached to the DOM, in case
+		// it was removed since the last `measure` call.
+		if ( targetElement && targetElement.isConnected ) {
```

Comment:
This is a separate fix for a bug that I discovered while working on this PR. Basically, the `measure` function could go on an infinite loop in case the `targetElement` was holding a reference to an old node that in the meantime was removed from the DOM. The `isConnected` check makes sure that we can detect that edge case and act correctly.

An example of this:
- in dev mode, react renders twice for the first render
- this was causing the `instanceId` of the whole `Tabs` to update, causing also the individual `Tabs.Tab` buttons to update.
- therefore, this `measure` function would loop indefinitely on first render